From 4a20469d0a85667a4867ea39ba7e0acdd090c038 Mon Sep 17 00:00:00 2001
From: RajaniRanjan <rajani.ranjan@intel.com>
Date: Wed, 16 Feb 2022 09:46:22 +0530
Subject: [PATCH] add input support for mulit-APP issue fixing

setting input coordinates according to display resolution

Tracked-On: OAM-100983
Signed-off-by: RajaniRanjan <rajani.ranjan@intel.com>
---
 client/src/vInputClient.c | 18 ++++++++++++------
 1 file changed, 12 insertions(+), 6 deletions(-)

diff --git a/client/src/vInputClient.c b/client/src/vInputClient.c
index 70ab925c..13d4469f 100644
--- a/client/src/vInputClient.c
+++ b/client/src/vInputClient.c
@@ -72,6 +72,7 @@ bool vinput_touch(struct vinputEvent ev){
 	if (mqId == -1) {
 		return false;
 	}
+u_int32_t temp;	
 	switch(ev.keys){
                 case LG_LEFT_ARROW:
                         ev.keys = LINUX_LEFT;
@@ -102,9 +103,13 @@ bool vinput_touch(struct vinputEvent ev){
                             ev.x = 0;}
                         else{
                             ev.x -= ev.dstRectx;}
-			ev.x *= XRES_MAX/ev.dstRectw;
-			ev.y *= XRES_MAX/ev.dstRectw;
-			ev.y += LTRB_Y;
+			temp = ev.x;
+			ev.x = ev.y;
+			ev.y = temp;
+                        ev.x *= XRES_MAX/ev.dstRecth;
+			ev.x = XRES_MAX - ev.x;
+                        ev.y *= YRES_MAX/ev.dstRectw;
+
 			break;
 
 		    case 2:
@@ -127,9 +132,10 @@ bool vinput_touch(struct vinputEvent ev){
                 switch (ev.rotation) {
                     case 1:
                     case 3:
-                        ev.x *= XRES_MAX/YRES_MAX;
-                        ev.y *= XRES_MAX/YRES_MAX;
-                        ev.y += LTRB_Y;
+			temp = ev.x;
+                        ev.x = ev.y;
+                        ev.y = temp;
+                        ev.x = XRES_MAX - ev.x;
                         break;
 
                     case 2:
-- 
2.25.1

